src/main/java/bio/terra/cda/app/util/NestedColumn.java

```diff
@@ -22,20 +25,20 @@ public static NestedColumn generate(String qualifiedColumnName) {
      */
     public static NestedColumn generate(String qualifiedColumnName)
         throws IllegalArgumentException {
-      StringBuilder unnestClause = new StringBuilder();
+      Set<String> unnestClauses = new LinkedHashSet<String>();
       String newColumn = null;
       if (qualifiedColumnName != null) {
         String[] c = qualifiedColumnName.split("\\.");
         if (c.length > 1) {
           newColumn = "_" + c[c.length - 2] + "." + c[c.length - 1];
-          unnestClause.append(", UNNEST(" + c[0] + ") AS _" + c[0]);
+          unnestClauses.add(", UNNEST(" + c[0] + ") AS _" + c[0]);
           for (int n = 1; n < c.length - 1; n++) {
-            unnestClause.append(", UNNEST(_" + c[n - 1] + "." + c[n] + ") AS _" + c[n]);
+            unnestClauses.add(", UNNEST(_" + c[n - 1] + "." + c[n] + ") AS _" + c[n]);
           }
-          return new NestedColumn(newColumn, unnestClause.toString());
+          return new NestedColumn(newColumn, unnestClauses);
         }
         if (c.length == 1) {
-          return new NestedColumn(qualifiedColumnName, "");
+          return new NestedColumn(qualifiedColumnName, null);
```

Comment:
Instead of `null`, return `Set.of()`. That way a caller doesn't need to check for null, so you can remove the `if (unnestClauses != null) {` from `testGeneratedUnnestClause()`.

Reply:
Used suggestion.

Reply:
Oops,  Set.of() is immutable,   I still need to add the unnest clauses for the WHERE clause.  I'll instantiate a new Set() because the the system value will always have an unnest clause that needs to be added.

Reply:
Ok, thanks for trying it out.